src/toil/jobStores/aws/jobStore.py

```diff
@@ -377,49 +419,53 @@ def delete_job(self, job_id):
         logger.debug("Deleting job %s", job_id)
 
         # If the job is overlarge, delete its file from the filestore
-        item = None
         for attempt in retry_sdb():
             with attempt:
-                item = self.jobsDomain.get_attributes(compat_bytes(job_id), consistent_read=True)
+                attributes = self.db.get_attributes(DomainName=self.jobsDomainName,
+                                                    ItemName=compat_bytes(job_id),
+                                                    ConsistentRead=True).get("Attributes", [])
         # If the overlargeID has fallen off, maybe we partially deleted the
         # attributes of the item? Or raced on it? Or hit SimpleDB being merely
         # eventually consistent? We should still be able to get rid of it.
-        self._checkItem(item, enforce = False)
-        if item.get("overlargeID", None):
+        self._checkAttributes(attributes, enforce=False)
+        overlarge_id_value = next((attribute["Value"] for attribute in attributes if attribute["Name"] == "overlargeID"), None)
```

Comment:
I'm getting tired of reading basically the same comprehension repeatedly. This could be wrapped in a function to save code, I think.